dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -559,69 +561,79 @@ public ListenableFuture<LoadFileResponse> load(
       // These two need UFS api support and cannot be achieved in a generic UFS interface.
       // We may be able to solve this by providing specific implementations for certain UFSes
       // in the future.
-      if (!loadData || !status.isFile()) {
-        continue;
-      }
-      long fileLength = status.asUfsFileStatus().getContentLength();
-      boolean countAsSkipped =
-          skipIfExists && isAllPageCached(status.asUfsFileStatus(), originalFs);
-      if (countAsSkipped) {
-        skippedFiles.incrementAndGet();
-        skippedFileLength.addAndGet(status.asUfsFileStatus().getContentLength());
-        continue;
-      }
-      if (fileLength == 0) {
-        continue;
-      }
       try {
-        ListenableFuture<Void> loadFuture = Futures.submit(() -> {
-          try {
-            if (options.hasUser()) {
-              AuthenticatedClientUser.set(options.getUser());
+        mMetaManager.put(ufsFullPath, fs);
+      } catch (Exception e) {
+        LOG.error("Failed to put file status to meta manager", e);
+        AlluxioRuntimeException t = AlluxioRuntimeException.from(e);
+        errors.add(LoadFailure.newBuilder().setUfsStatus(status.toProto())
+                                  .setCode(t.getStatus().getCode().value()).setRetryable(true)
+                                  .setMessage(t.getMessage()).build());
+      }
+    }
+    for (Block block : blocks) {
+      if (block.getLength() > 0) {
+        boolean countAsSkipped = skipIfExists && isAllPageCached(block);
+        if (countAsSkipped) {
+          numSkipped.incrementAndGet();
+          skippedLength.addAndGet(block.getLength());
+          continue;
+        }
+        try {
+          ListenableFuture<Void> loadFuture = Futures.submit(() -> {
+            try {
+              if (options.hasUser()) {
+                AuthenticatedClientUser.set(options.getUser());
+              }
+              loadData(block.getUfsPath(), 0, block.getOffsetInFile(), block.getLength(),
```

Comment:
we dont need the mountid 0 any more do we?

Reply:
probably we need to integrate this with multi mount point change so we still need to input something later on? @dbw9580